### PR TITLE
Add iothub API group

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Fixes https://github.com/upbound/sales/issues/8

Following Terraform resources are configured:
  azurerm_iothub
  azurerm_iothub_consumer_group
  azurerm_iothub_dps
  azurerm_iothub_dps_certificate
  azurerm_iothub_dps_shared_access_policy
  azurerm_iothub_shared_access_policy

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.
- [x] Author example manifest(s) at least for the IOTHub resource

### How has this code been tested

Tested via the example manifest provided.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by ulucinar** [comment]
I need to figure out:
https://github.com/crossplane-contrib/provider-jet-azure/blob/a6395803c5e81e592d436a85a94f0a687ac8a3d3/config/iothub/config.go#L39

**Comment by ulucinar** [comment]
> I need to figure out:
> 
> https://github.com/crossplane-contrib/provider-jet-azure/blob/a6395803c5e81e592d436a85a94f0a687ac8a3d3/config/iothub/config.go#L39

Resolved this issue.

File: config/iothub/config.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+const (
+	groupIOTHub = "iothub"
+)
+
+// Configure configures iothub group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_iothub", func(r *config.Resource) {
+		r.Kind = "IOTHub"
+		r.ShortGroup = groupIOTHub
```
Comment (turkenh) [inline config/iothub/config.go:34]: nit: we might consider overriding default group for all iothub resources in default resource, like: https://github.com/crossplane-contrib/provider-jet-gcp/blob/ca147c40ebe349cb933396624890bc9b92745da8/config/overrides.go#L14

**Review by turkenh — APPROVED** [review]
Looking great, thanks @ulucinar 👍

File: config/iothub/config.go (hunk 1 above)
Comment (ulucinar) [reply config/iothub/config.go]: Thanks @turkenh! Done.

**Pull request merged**